### DISCUSSION TITLE
Red colored messages when starting labwc from tty

### DISCUSSION THREAD

**Discussion by moilong** [labwc/labwc#1162] [General]
labwc 0.6.5-7-ge5aef03. No issues in running labwc but Im getting this messages everytime upon starting.

00:00:00.001 [../src/config/rcxml.c:191] window-switcher field 'app_id' is deprecated
00:00:00.001 [../src/config/rcxml.c:279] expect <action name=""> element first. nodename: 'direction.action' content: 'left'
00:00:00.001 [../src/config/rcxml.c:279] expect <action name=""> element first. nodename: 'direction.action' content: 'right'
00:00:00.001 [../src/config/rcxml.c:279] expect <action name=""> element first. nodename: 'direction.action' content: 'up'
00:00:00.001 [../src/config/rcxml.c:279] expect <action name=""> element first. nodename: 'direction.action' content: 'down'
00:00:00.002 [../src/config/rcxml.c:339] expect <action name=""> element first. nodename: 'menu.action' content: 'root-menu'
00:00:00.002 [../src/config/rcxml.c:339] expect <action name=""> element first. nodename: 'menu.action' content: 'root-menu'
amdgpu: amdgpu_cs_ctx_create2 failed. (-13)
00:00:00.545 [types/wlr_drm_lease_v1.c:715] No DRM backend supplied, failed to create wlr_drm_lease_v1_manager

**Comment by stefonarch** [comment]
> 00:00:00.001 [../src/config/rcxml.c:191] window-switcher field 'app_id' is deprecated

`man labwc-config`
```
    <windowSwitcher><fields><field content="" width="%">
           Define window switcher fields.

           content defines what the field shows and can be any of:

               •   type Show view type ("xdg-shell" or "xwayland")

               •   identifier Show identifier (app_id for native Wayland windows and WM_CLASS for XWayland clients)

               •   title Show window title if different to app_id
```

Post the other lines mentioned in the messages.

**Reply by moilong** [reply]
[rcxml.txt](https://github.com/labwc/labwc/files/12906602/rcxml.txt)
Those are the only lines. I did not get any other messages. Attaching my rc.xml just in case.

**Reply by johanmalm** [reply]
Use identifier instead of app_id

**Comment by stefonarch** [comment]
Sorry, my bad, thought the errors listed the lines in` rc.xml`. 
Searching for left I guess those are the lines, just remove them to disactivate them.

 ```
 <keybind key="A-Left">
      <action name="None" direction="left" />
    </keybind>
    <keybind key="A-Right">
      <action name="None" direction="right" />
    </keybind>
    <keybind key="A-Up">
      <action name="None" direction="up" />
    </keybind>
    <keybind key="A-Down">
      <action name="None" direction="down" />
    </keybind>


      <mousebind button="Left" action="Press">
        <action name="None" menu="root-menu" />
      </mousebind>
      <mousebind button="Right" action="Press">
        <action name="None" menu="root-menu" />
      </mousebind>


```

**Reply by Consolatis** [reply]
To remove all `A-cursor` keybinds the following should do the same:
```xml
<keybind key="A-Left" />
<keybind key="A-Right" />
<keybind key="A-Up" />
<keybind key="A-Down" />
```

Alternatively you can just remove the `direction` argument from the `None` actions, the result should be the same.

**Comment by moilong** [comment]
Thanks everybody. Removed those lines in rc.xml as @stefonarch said and now only the message about DRM backend is showing.
amdgpu: amdgpu_cs_ctx_create2 failed. (-13)
00:00:00.545 [types/wlr_drm_lease_v1.c:715] No DRM backend supplied, failed to create wlr_drm_lease_v1_manager

**Reply by Consolatis** [reply]
The message itself is harmless (unless you want to make use of the protocol for VR headsets or similar).
I am not sure why it is even shown in this case though (it originates from wlroots), when starting labwc from the TTY it should in fact use the DRM backend.

Edit:
Just looked at the wlroots code: it tries to create a "lease device" and if that fails it complains about not running with a DRM backend (which is wrong). I assume the reason it fails is `amdgpu: amdgpu_cs_ctx_create2 failed. (-13)` (whatever that means).

**Reply by moilong** [reply]
Thanks for the details.
